api/src/main/java/org/bahmni/module/bahmni/ie/apps/service/impl/BahmniFormServiceImpl.java

```diff
@@ -174,6 +175,31 @@ public ExportResponse getFormsByListOfUuids(List<String> formUuids) {
         return new ExportResponse(bahmniFormDataList, errorFormNames);
     }
 
+    @Override
+    public BahmniFormResource saveFormNameTranslation(BahmniFormResource bahmniFormResource, String referenceFormUuid) {
+        Form form = formService.getFormByUuid(bahmniFormResource.getForm().getUuid());
+        FormResource formResource = getFormResource(bahmniFormResource.getUuid());
+        formResource.setForm(form);
+        formResource.setName(bahmniFormResource.getForm().getName() + "_FormName_Translation");
+        formResource.setDatatypeClassname(FormNameTranslationDatatype.class.getName());
+        if (referenceFormUuid == null || referenceFormUuid.trim().equals("")) {
+            formResource.setDatatypeConfig(bahmniFormResource.getValue());
+            formResource.setValue(bahmniFormResource.getValue());
+        } else {
+            String value = grtOldFormResourceValue(referenceFormUuid);
```

Comment:
@vinay-ThoughtWorks is it getOldFormResourceValue() or grtOldFormResourceValue() ??

Reply:
It's a typo.Will fix.

Reply:
Done